### PR TITLE
fix(pubsub): Filter displayed list of Pub/Sub Subscription Names by Pub/Sub System Type.

### PR DESCRIPTION
**Pull request by joonlim**
* Part of a bug fix to display only Pub/Sub subscription names of subscriptions associated with a given Pub/Sub system type (e.g., 'google').
* This fix also touches echo and gate.
* This PR also removes 'kafka' from list of Pub/Sub System Types in settings.js and pubsub.trigger.ts as it is not being used.
* We have opted out of adding a button to refresh the cached list of subscriptions because @danielpeach deemed it an unlikely use case.

### CODE REVIEW

**Comment by joonlim** [comment]
@jtk54 @danielpeach Could you please review?

File: app/scripts/modules/core/src/pipeline/config/triggers/pubsub/pubsub.trigger.ts [hunk 1]
```diff
@@ -8,8 +8,9 @@ import { IPubsubTrigger } from 'core/domain';
 import { ServiceAccountService } from 'core';
 
 class PubsubTriggerController implements IController {
-  public pubsubSystems = SETTINGS.pubsubProviders || ['google', 'kafka'];
-  public pubsubSubscriptions: string[];
+  public pubsubSystems = SETTINGS.pubsubProviders || ['google']; // TODO: Add amazon once it is confirmed that amazon pub/sub works.
```
Comment (jtk54) [inline app/scripts/modules/core/src/pipeline/config/triggers/pubsub/pubsub.trigger.ts]: Add your name to the TODO :)

File: app/scripts/modules/core/src/pubsub/pubsubSubscription.service.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import { module, IPromise } from 'angular';
 
 import { API } from 'core/api/ApiService';
+import { IPubsubSubscription } from 'core/domain';
```
Comment (jtk54) [inline app/scripts/modules/core/src/pubsub/pubsubSubscription.service.ts]: Both this import and the one above can be imported from '@spinnaker/core'.

File: settings.js [hunk 3]
```diff
@@ -152,7 +152,7 @@ window.spinnakerSettings = {
   authEnabled: authEnabled,
   authTtl: 600000,
   gitSources: ['stash', 'github', 'bitbucket'],
-  pubsubProviders: ['google', 'kafka'],
+  pubsubProviders: ['google'], // TODO(joonlim): Add amazon once it is confirmed that amazon pub/sub works.
```
Comment (danielpeach) [inline settings.js]: I think there should be a corresponding change in `.halconfig/settings.js`.

Comment (joonlim) [reply settings.js]: Fixed. Looks like my editor auto-linted ```halconfig/settings.js```. Is this desired or should I restore those changes?

Comment (jtk54) [reply settings.js]: Leaving the linted changes makes sense to me, shouldn't matter much.

**Pull request merged**